### PR TITLE
Add circleci and travisci

### PR DESCRIPTION
**Pull request by andelf**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
+#  - osx
+
+before_script:
+- cargo install rustfmt
 
 script:
-  - rustc -Z parse-only src/lib.rs
+- cargo fmt && git diff-index --quiet HEAD -- || (echo please make format and run tests before creating a pr!; exit 1)
+- cargo build --features static-link
+- cargo test --all --features static-link
```
Comment (siddontang) [inline .travis.yml]: do we need to build RocksDB every time?

Comment (andelf) [inline .travis.yml]: or removing travis-ci.
using circle-ci is enough.

Comment (hhkbp2) [inline .travis.yml]: @andelf Why do we need two CI systems at the first place?

**Comment by BusyJay** [comment]
Do we need to support OSX too?

**Comment by siddontang** [comment]
For a library, I suggest supporting stable, beta and nightly version.

**Comment by siddontang** [comment]
CI failed @andelf

**Comment by siddontang** [comment]
CI still failed @andelf

**Comment by andelf** [comment]
how to setup rustfmt version?
CI fails because 0.8.3 conflicts with 0.6.2.

**Comment by siddontang** [comment]
can we force using 0.6 version like TiKV does?

**Comment by arthurprs** [comment]
You could you add `#![deny(warnings)]` in the lib.rs, to fails the build if there're warnings.

**Comment by siddontang** [comment]
LGTM

PTAL @zhangjinpeng1987

**Comment by zhangjinpeng1987** [comment]
LGTM

**Pull request merged**